### DISCUSSION TITLE
Updating graphfilter plugin to 0.9.3

### DISCUSSION THREAD

**Discussion by ypanagis** [gephi/gephi-plugins#247] [Q&A] [answered]
I am trying to update the plugin for 0.9.3 and seeing some issues. Sorry if those are perhaps straightforward but my Java has been left a bit behind. So here are some issues:
1. After forking the `gephi-plugins` version 0.9.3 by cloning via `https`, I updated the pom file in both `gephi-plugins-parent` and `gephi-plugins`. Both `mvn clean package` and NetBeans "Clean and Build" (which actually runs `mvn clean install`), give this error:

```
[ERROR] Failed to execute goal on project graphfilter: Could not resolve dependencies for project dk.icourts:graphfilter:nbm:1.0.0: Failed to collect dependencies at org.gephi:graph-api:jar:0.9.3 -> org.gephi:project-api:jar:0.9.3 -> org.netbeans.api:org-openide-filesystems:jar:RELEASE81: Failed to read artifact descriptor for org.netbeans.api:org-openide-filesystems:jar:RELEASE81: Could not transfer artifact org.netbeans.api:org-openide-filesystems:pom:RELEASE81 from/to netbeans (http://bits.netbeans.org/nexus/content/groups/netbeans/): authentication failed for http://bits.netbeans.org/nexus/content/groups/netbeans/org/netbeans/api/org-openide-filesystems/RELEASE81/org-openide-filesystems-RELEASE81.pom, status: 401 Repository decommissioned. Please refer to https://netbeans.apache.org/about/oracle-transition.html for more information. -> [Help 1]
```
2. If I change `packaging` to `pom` in the `pom.xml` in `[graphfilter]`, I get the message 
`Plugin org.apache.netbeans.utilities:nbm-maven-plugin:3.14 or one of its dependencies could not be resolved: Could not find artifact org.apache.netbeans.utilities:nbm-maven-plugin:jar:3.14 in oss-sonatype (https://oss.sonatype.org/content/repositories/snapshots/) -> [Help 1]`. It seems though that my module, `filterfromfile` is being built successfully with a warning for nbm-maven-plugin: `The POM for org.apache.netbeans.utilities:nbm-maven-plugin:jar:3.14 `. I am attaching the [maven output](https://github.com/gephi/gephi-plugins/files/8553952/mvn.message.txt).  and hopefully it helps more.

**Comment by mbastian** [comment, marked as answer]
Hi @ypanagis 

Thanks for the log message (it always help). Alternatively, you could also commit to your fork so I could directly look at the code and try myself.

It seems that there is still some references to old repositories or old plugin versions. NBM Plugin version 3.14 doesn't sound right as we should be at [4.5](https://github.com/gephi/gephi-plugins/blob/a70acdd70ff6187988f373c4baf3f19c11c94637/pom.xml#L87). My hypothesis is that you parent POM is outdated and is messing up things. In fact, you no longer need to have the parent pom locally as [we publish it to Maven central](https://search.maven.org/artifact/org.gephi/gephi-plugin-parent/0.9.3/pom). In other words, delete the file `modules/pom.xml` and try again. Make sure your plugin module `filterfromfile` has the proper parent pom: 
```
<parent>
        <artifactId>gephi-plugin-parent</artifactId>
        <groupId>org.gephi</groupId>
        <version>0.9.3-SNAPSHOT</version>
</parent>
```

**Reply by ypanagis** [reply]
Hi @mbastian ,

Thank you for the suggestion, I made the changes you suggested but now it complains for _non-resolvable parent POM_ and that it cannot find `relativePath` for the parent:

```
[ERROR] Some problems were encountered while processing the POMs:
Non-resolvable parent POM for dk.icourts:graphfilter:1.0.0: Could not find artifact org.gephi:gephi-plugin-parent:pom:0.9.3-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 4, column 13
 @ 
The build could not read 1 project -> [Help 1]
  
  The project dk.icourts:graphfilter:1.0.0 (C:\Users\dcg601\git\gephi-plugins-new\modules\filterfromfile\pom.xml) has 1 error
    Non-resolvable parent POM for dk.icourts:graphfilter:1.0.0: Could not find artifact org.gephi:gephi-plugin-parent:pom:0.9.3-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 4, column 13 -> [Help 2]
```

I have actually pushed the changes in my fork in case you could have a quick look and cure the patient :-)

**Reply by mbastian** [reply]
Cool, I was able to browse the code. A few things you need to correct:
- Use version 0.9.3, not 0.9.3-SNAPSHOT. The parent POM can't be found otherwise.
- You still use the old Netbeans NBM plugin. Make sure yo update this line: https://github.com/ypanagis/gephi-plugins/blob/00027ff6916a47e6c80a18eb06595df83979fc67/modules/filterfromfile/pom.xml#L43
- Make sure to have `<packaging>nbm</packaging>` in your POM file.
- You still seem to be using `<classifier>linux</classifier>` in the root POM while you should be using `<classifier>linux-x64</classifier>`. A couple of other things seem outdated in that file. I suggest you simply [copy from master ](https://github.com/gephi/gephi-plugins/blob/master/pom.xml) and add your module again.

That should fix all the issues hopefully

**Comment by ypanagis** [comment]
Building the plugin after the modifications you suggested, _the plugin is being built_ but building Gephi fails with the message:

`[ERROR] Failed to execute goal org.gephi:gephi-maven-plugin:1.2.3:validate (validate-plugin) on project gephi-plugins: The 'licenseName' configuration should be set for the project 'filterfromfile'. This can be added to the configuration of the 'nbm-maven-plugin' plugin. In addition, a 'licenseFile' can be specified, relative to the module's root folder. -> [Help 1]`

My `pom.xml` however contains `licenseName` as can be seen here:
```           
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>nbm-maven-plugin</artifactId>
                <configuration>
                    <author>Ioannis Panagis</author>
                    <sourceCodeUrl>https://github.com/ypanagis/gephi-plugins</sourceCodeUrl>
                    <licenseName>Apache 2 GPL</licenseName>
                    <licenseFile>apachegpl2.txt</licenseFile>
                    <homePageUrl>http://jura.ku.dk/icourts/staff</homePageUrl>
                </configuration>
            </plugin>
```

The NBM file works, though, and _can be imported by Gephi_. Would you suggest to commit this version?

**Reply by mbastian** [reply]
Cool, but you're still missing Step 3 from [Maintainers: Update Plugins for Gephi 0.9.3 ](https://github.com/gephi/gephi-plugins/discussions/245).

```
<groupId>org.apache.netbeans.utilities</groupId>
<artifactId>nbm-maven-plugin</artifactId>
```

**Reply by mbastian** [reply]
And your `gephi-maven-plugin` version is still `1.2.3` instead of `1.3.0`. Once again, I suggest you copy from master the `pom.xml`.

**Comment by ypanagis** [comment]
Done the above, pushed and created a pull request but `build_and_test` fails with `Error: Failed to execute goal org.gephi:gephi-maven-plugin:1.3.0:validate (validate-plugin) on project gephi-plugins: Multiple module suites have been found, this is not supported -> [Help 1]`.
Both the module and Gephi get correctly built, with `mvn org.gephi:gephi-maven-plugin:run`, in my computer.

**Reply by ypanagis** [reply]
@mbastian I hope it is ok to ping you. I just committed the parent `pom.xml` that has the correct version but can't see of course if the building can proceed now... It was a mess I created but I hope PR #249 could be built and approved and I apologise again.

**Reply by mbastian** [reply]
Thanks, your plugin was migrated automatically for 0.9.3 so there wasn't any need to do a PR in fact, but all good your plugin should be usable.
